qiskit/circuit/annotated_operation.py

```diff
@@ -219,6 +220,29 @@ def power(self, exponent: float, annotated: bool = False):
         extended_modifiers.append(PowerModifier(exponent))
         return AnnotatedOperation(self.base_op, extended_modifiers)
 
+    @property
+    def params(self) -> list[ParameterValueType]:
+        """The params of the underlying base operation."""
+        return getattr(self.base_op, "params", [])
+
+    @params.setter
+    def params(self, value: list[ParameterValueType]):
+        if hasattr(self.base_op, "params"):
+            self.base_op.params = value
+        else:
+            raise NotImplementedError(
+                f"Cannot set attribute ``params`` on the base operation {self.base_op}."
```

Comment:
I am wondering if there might be a better error message type than `NotImplementedError`.

Reply:
Maybe `AttributeError`?

Reply:
AttributeError sounds about right:

```suggestion
            raise AttributeError(
                f"Cannot set attribute ``params`` on the base operation {self.base_op}."
```